dimagi/commcare-hq
# Use Mapbox Layer for Cases and User Markers in Microplanning Map

## Product Description

The markers themselves will now look slightly different as shown below:

**Old:**
![Screenshot from 2024-12-18 12-16-08](https://github.com/user-attachments/assets/887013b4-5a14-4543-b801-47f67d8e69b8)

**New:**
![Screenshot from 2024-12-18 12-15-17](https://github.com/user-attachments/assets/2e57c35b-32c6-4dd2-ae38-cd40e4242d16)

## Technical Summary

Link to ticket [here](https://dimagi.atlassian.net/browse/SC-4104).
Related tech spec doc can be found [here](https://docs.google.com/document/d/1tpn8Q8SP2pQnLsPtn9kfcuJawzaoER4jK-AxAtAzksg/edit?tab=t.0).

This PR introduces a core change to the Microplanning Map page in the form of completely changing how markers are loaded onto the map. Previously, the markers were created as separate HTML elements which were then added onto the map, however this caused performance issues if there were a lot of markers on the map (e.g. 1000) and ultimately limited the scalability of this feature.

The marker logic has been reworked to use a single Mapbox data source with two layers, one for users and one for cases. The markers themselves are shown on the map with a simple image which gets loaded in when initializing the map. Using a map layer instead of individual HTML elements for the markers allows the page to achieve greater performance which, in turn, will allow for displaying more markers on the map at any given time.

In addition to the above implementation, this PR also resolves the following two bugs on the Microplanning Map page:
- Reviewing assignment results with polygon filtering applied resulted in a JS exception being thrown in the browser.
- Running a new disbursement does not remove the previous disbursement results from the assignment popup.

## Feature Flag

`MICROPLANNING`

## Safety Assurance

### Safety story

- Local testing
- QA has been completed

End-to-end testing was done locally for both the Microplanning Map and Case Clustering Map pages to ensure that all functionality still works as expected. Additionally, QA will be doing end-to-end testing on these two pages as well to make sure all features still work.

### Automated test coverage


N/A

### QA Plan


QA has been completed. Ticket can be viewed [here](https://dimagi.atlassian.net/browse/QA-7355).

### Rollback instructions




### Labels & Review
